backtester/funding/trackingcurrencies/trackingcurrencies_test.go

```diff
@@ -0,0 +1,222 @@
+		t.Errorf("received '%v' expected '%v'", err, nil)
+	}
+	if len(resp) != 3 {
+		t.Error("expected 3 currency settings as it did not contain a USD equiv")
+	}
+}
+
+func TestFindMatchingUSDPairs(t *testing.T) {
+	t.Parallel()
+	type testPair struct {
+		description    string
+		initialPair    currency.Pair
+		availablePairs *currency.PairStore
+		basePair       currency.Pair
+		quotePair      currency.Pair
+		expectedErr    error
+	}
+	tests := []testPair{
+		{
+			description:    "already has USD",
+			initialPair:    currency.NewPair(currency.BTC, currency.USDT),
+			availablePairs: &currency.PairStore{Available: currency.Pairs{currency.NewPair(currency.BTC, currency.USDT)}},
+			basePair:       currency.Pair{},
+			quotePair:      currency.Pair{},
+			expectedErr:    ErrCurrencyContainsUSD,
+		},
+		{
+			description:    "successful",
+			initialPair:    currency.NewPair(currency.BTC, currency.LTC),
+			availablePairs: &currency.PairStore{Available: currency.Pairs{currency.NewPair(currency.BTC, currency.LTC), currency.NewPair(currency.BTC, currency.USDT), currency.NewPair(currency.LTC, currency.TUSD)}},
+			basePair:       currency.NewPair(currency.BTC, currency.USDT),
+			quotePair:      currency.NewPair(currency.LTC, currency.TUSD),
+			expectedErr:    nil,
+		},
+		{
+			description:    "quote currency has no matching USD pair",
+			initialPair:    currency.NewPair(currency.BTC, currency.LTC),
+			availablePairs: &currency.PairStore{Available: currency.Pairs{currency.NewPair(currency.BTC, currency.LTC), currency.NewPair(currency.BTC, currency.DAI)}},
+			basePair:       currency.NewPair(currency.BTC, currency.DAI),
+			quotePair:      currency.Pair{},
+			expectedErr:    errNoMatchingQuoteUSDFound,
+		},
+		{
+			description:    "base currency has no matching USD pair",
+			initialPair:    currency.NewPair(currency.BTC, currency.LTC),
+			availablePairs: &currency.PairStore{Available: currency.Pairs{currency.NewPair(currency.BTC, currency.LTC), currency.NewPair(currency.LTC, currency.USDT)}},
+			basePair:       currency.Pair{},
+			quotePair:      currency.NewPair(currency.LTC, currency.USDT),
+			expectedErr:    errNoMatchingBaseUSDFound,
+		},
+		{
+			description:    "both base and quote don't have USD pairs",
+			initialPair:    currency.NewPair(currency.BTC, currency.LTC),
+			availablePairs: &currency.PairStore{Available: currency.Pairs{currency.NewPair(currency.BTC, currency.LTC)}},
+			basePair:       currency.Pair{},
+			quotePair:      currency.Pair{},
+			expectedErr:    errNoMatchingPairUSDFound,
+		},
+		{
+			description:    "currency doesnt exist in available pairs",
+			initialPair:    currency.NewPair(currency.BTC, currency.LTC),
+			availablePairs: &currency.PairStore{Available: currency.Pairs{currency.NewPair(currency.BTC, currency.DOGE)}},
+			basePair:       currency.Pair{},
+			quotePair:      currency.Pair{},
+			expectedErr:    errCurrencyNotFoundInPairs,
+		},
+	}
+	for i := range tests {
```

Comment:
```suggestion
	for i := range tests {
		tt := tests[i]
		t.Run(tt.description, func(t *testing.T) {
			t.Parallel()
			basePair, quotePair, err := findMatchingUSDPairs(tt.initialPair, tt.availablePairs)
			if !errors.Is(err, tt.expectedErr) {
				t.Fatalf("'%v' received '%v' expected '%v'", tt.description, err, tt.expectedErr)
			}
			if basePair != tt.basePair {
				t.Fatalf("'%v' received '%v' expected '%v'", tt.description, basePair, tt.basePair)
			}
			if quotePair != tt.quotePair {
				t.Fatalf("'%v' received '%v' expected '%v'", tt.description, quotePair, tt.quotePair)
			}
		})
	}
```

As individually testing this func does not provide coverage